# Exploring dataset

In [1]:
import numpy as np
import pandas as pd

In [4]:
demographics_cvs_path = 'MatchedControls/demographics.csv'
subject_file = 'src/new/subjects.txt'

In [5]:
demographics = pd.read_csv(demographics_cvs_path)

In [32]:
demographics

Subject Sex    Age Site
0      R0203_V0390   F   8.08  TOB
1      R0077_V0154   F   8.92  TOB
2      R0100_V0195   F   9.08  TOB
3      R0110_V0211   F   9.08  TOB
4      R0253_V0511   F   9.33  TOB
..             ...  ..    ...  ...
190  113223_005639   M  16.67  PNC
191  089834_003353   M  16.83  PNC
192  085960_003323   M  17.25  PNC
193  125280_005515   M  17.42  PNC
194  120514_005912   M  17.58  PNC

[195 rows x 4 columns]

In [33]:
demographics['Site'].value_counts()

PNC    98
TOB    97
Name: Site, dtype: int64

In [31]:
demographics['Sex'].value_counts()

M    153
F     42
Name: Sex, dtype: int64

In [19]:
gb = demographics.groupby('Site')
gb.describe()

Age                                                         
     count       mean       std   min      25%    50%    75%    max
Site                                                               
PNC   98.0  12.339490  2.708077  8.17  10.1025  12.00  14.33  18.08
TOB   97.0  12.241959  2.737532  8.00  10.0800  12.08  14.25  17.83

In [39]:
gb_sex = demographics.groupby('Sex')
gb_sex.describe()

Age                                                           
     count       mean       std   min      25%    50%      75%    max
Sex                                                                  
F     42.0  12.096667  3.035141  8.08  10.1025  11.08  14.4575  18.08
M    153.0  12.344314  2.630222  8.00  10.0800  12.25  14.2500  17.75

In [46]:
demographics.groupby(['Site', 'Sex']).describe()

Age                                                           
         count       mean       std   min      25%    50%      75%    max
Site Sex                                                                 
PNC  F    21.0  12.174286  3.080129  8.17  10.1700  11.08  14.3300  18.08
     M    77.0  12.384545  2.617989  8.17  10.0800  12.25  14.3300  17.58
TOB  F    21.0  12.019048  3.063482  8.08  10.0800  11.25  14.6700  17.83
     M    76.0  12.303553  2.659328  8.00  10.1475  12.21  14.1225  17.75

The connectome comes from 2 scanner sites, in which 98 is from PNC and 97 is from TOB. There are in total 153 males and 42 females subjects. As we can see, the number of males and females subjects are distributed equaly between 2 scanners. Each sites has 21 females and around 76 males subjects.

## 2. Parse scanner and subjects to txt file

In [68]:
PNC_F = demographics.loc[demographics['Site'] == 'PNC'][demographics['Sex'] == 'F']
PNC_M = demographics.loc[demographics['Site'] == 'PNC'][demographics['Sex'] == 'M']
TOB_F = demographics.loc[demographics['Site'] == 'TOB'][demographics['Sex'] == 'F']
TOB_M = demographics.loc[demographics['Site'] == 'TOB'][demographics['Sex'] == 'M']


/tmp/ipykernel_1301/1119866737.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  PNC_F = demographics.loc[demographics['Site'] == 'PNC'][demographics['Sex'] == 'F']
/tmp/ipykernel_1301/1119866737.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  PNC_M = demographics.loc[demographics['Site'] == 'PNC'][demographics['Sex'] == 'M']
/tmp/ipykernel_1301/1119866737.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  TOB_F = demographics.loc[demographics['Site'] == 'TOB'][demographics['Sex'] == 'F']
/tmp/ipykernel_1301/1119866737.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  TOB_M = demographics.loc[demographics['Site'] == 'TOB'][demographics['Sex'] == 'M']


In [70]:
file_map = {
    'PNC_F': PNC_F,
    'PNC_M': PNC_M,
    'TOB_F': TOB_F,
    'TOB_M': TOB_M
}

for scanner_subject_name, df in file_map.items():
    with open(f'src/new/{scanner_subject_name}_subjects.txt', 'w') as f:
        for s in df['Subject'].tolist():
            f.write("%s\n" % s)